core/micrometer/src/main/java/org/eclipse/dataspaceconnector/micrometer/MicrometerExtension.java

```diff
@@ -0,0 +1,90 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.micrometer;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.binder.jvm.ClassLoaderMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmGcMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmMemoryMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmThreadMetrics;
+import io.micrometer.core.instrument.binder.okhttp3.OkHttpMetricsEventListener;
+import io.micrometer.core.instrument.binder.system.ProcessorMetrics;
+import okhttp3.EventListener;
+import org.eclipse.dataspaceconnector.core.executor.ExecutorInstrumentationImplementation;
+import org.eclipse.dataspaceconnector.spi.EdcSetting;
+import org.eclipse.dataspaceconnector.spi.system.BaseExtension;
+import org.eclipse.dataspaceconnector.spi.system.Provides;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+
+@BaseExtension
+@Provides({EventListener.class, ExecutorInstrumentationImplementation.class})
+public class MicrometerExtension implements ServiceExtension {
+
+    @EdcSetting
+    public static final String ENABLE_METRICS = "metrics.enabled";
+    @EdcSetting
+    public static final String ENABLE_SYSTEM_METRICS = "metrics.system.enabled";
+    @EdcSetting
+    public static final String ENABLE_OKHTTP_METRICS = "metrics.okhttp.enabled";
+    @EdcSetting
+    public static final String ENABLE_EXECUTOR_METRICS = "metrics.executor.enabled";
+
+    @Override
+    public String name() {
+        return "Micrometer Metrics";
+    }
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        MeterRegistry registry = Metrics.globalRegistry;
+        context.registerService(MeterRegistry.class, registry);
+
+        var enableMetrics = Boolean.valueOf(context.getSetting(ENABLE_METRICS, "true"));
+        var enableSystemMetrics = Boolean.valueOf(context.getSetting(ENABLE_SYSTEM_METRICS, "true"));
+        var enableOkHttpMetrics = Boolean.valueOf(context.getSetting(ENABLE_OKHTTP_METRICS, "true"));
+        var enableExecutorMetrics = Boolean.valueOf(context.getSetting(ENABLE_EXECUTOR_METRICS, "true"));
+
+        if (enableMetrics && enableSystemMetrics) {
```

Comment:
why we always need enableMetrics ? is enableSystemMetrics/enableOkHttpMetrics/enableExecutorMetrics is not enough ?

Reply:
As discussed, leaving it as is.